src/Build/Evaluation/Evaluator.cs

```diff
@@ -250,7 +252,8 @@ internal class Evaluator<P, I, M, D>
             _sdkResolverService = sdkResolverService;
             _submissionId = submissionId;
             _evaluationContext = evaluationContext;
-            _evaluationProfiler = new EvaluationProfiler((loadSettings & ProjectLoadSettings.ProfileEvaluation) != 0);
+            _profileEvaluation = profileEvaluation;
+            _evaluationProfiler = new EvaluationProfiler(_profileEvaluation);
```

Comment:
Since the Evaluator.EvaluationProfiler field is always instantiated and it already knows whether it's on or off, the Evaluator._profileEvaluation field does not seem like it's needed. You could query the EvaluationProfiler instead.